src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolderDescriptor.java

```diff
@@ -75,8 +100,74 @@ public String getCategoryId() {
         return r;
     }
 
+    /**
+     * Gets the {@link FolderIconDescriptor}s applicable for this folder type.
+     *
+     * @since FIXME
+     */
+    public List<FolderIconDescriptor> getIconDescriptors() {
+        List<FolderIconDescriptor> r = new ArrayList<FolderIconDescriptor>();
+        for (FolderIconDescriptor p : FolderIconDescriptor.all()) {
+            if (p.isApplicable(clazz.asSubclass(AbstractFolder.class))) {
+                r.add(p);
+            }
+        }
+        StaplerRequest request = Stapler.getCurrentRequest();
+        if (request != null) {
+            AbstractFolder folder = request.findAncestorObject(AbstractFolder.class);
+            if (folder != null) {
+                return DescriptorVisibilityFilter.apply(folder, r);
+            }
+        }
+        return r;
+    }
+
+    // TODO remove once baseline 2.0
+    public String getDescription() {
+        Stapler stapler = Stapler.getCurrent();
+        if (stapler != null) {
+            try {
+                WebApp webapp = WebApp.getCurrent();
+                MetaClass meta = webapp.getMetaClass(this);
+                Script s = meta.loadTearOff(JellyClassTearOff.class).findScript("newInstanceDetail");
```

Comment:
> in order to not break descriptions in the case where somebody upgrades folders where the plugin that extends folders has not been updated

I think this was just rejected as an unnecessary defense—the impact is minor and people should have long since upgraded those plugins anyway.